pkg/suggestion/v1alpha3/goptuna/converter.go

```diff
@@ -0,0 +1,350 @@
+package suggestion_goptuna_v1alpha3
+
+import (
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/c-bata/goptuna"
+	"github.com/c-bata/goptuna/cmaes"
+	"github.com/c-bata/goptuna/tpe"
+	api_v1_alpha3 "github.com/kubeflow/katib/pkg/apis/manager/v1alpha3"
+)
+
+func toGoptunaDirection(t api_v1_alpha3.ObjectiveType) goptuna.StudyDirection {
+	if t == api_v1_alpha3.ObjectiveType_MINIMIZE {
+		return goptuna.StudyDirectionMinimize
+	}
+	return goptuna.StudyDirectionMaximize
+}
+
+func toGoptunaSampler(algorithm *api_v1_alpha3.AlgorithmSpec) (goptuna.Sampler, goptuna.RelativeSampler, error) {
+	name := algorithm.GetAlgorithmName()
+	if name == AlgorithmCMAES {
+		opts := make([]cmaes.SamplerOption, 0, len(algorithm.GetAlgorithmSetting())+1)
+		opts = append(opts, cmaes.SamplerOptionNStartupTrials(0))
+		for _, s := range algorithm.GetAlgorithmSetting() {
+			if s.Name == "random_state" {
+				seed, err := strconv.Atoi(s.Value)
+				if err != nil {
+					return nil, nil, err
+				}
+				opts = append(opts, cmaes.SamplerOptionSeed(int64(seed)))
+			} else if s.Name == "sigma" {
+				sigma, err := strconv.ParseFloat(s.Value, 64)
+				if err != nil {
+					return nil, nil, err
+				}
+				opts = append(opts, cmaes.SamplerOptionInitialSigma(sigma))
+			}
+		}
+		return nil, cmaes.NewSampler(opts...), nil
+	} else if name == AlgorithmTPE {
```

Comment:
I think we will have two suggestion providers which support TPE, do you think we should keep a prefix for it? Maybe gotuna-tpe?

/cc @andreyvelich @johnugeorge

Reply:
Agree, we can name it goptuna-tpe and goptuna-random.

Reply:
I'm a little bit concerned about the consistency of naming. It might be enough just updating `katib-config` if we want to use Goptuna suggestion service.

```
:
  suggestion: |-
    {
      "random": {
        "image": "gcr.io/kubeflow-images-public/katib/v1alpha3/suggestion-goptuna"
      },
      "tpe": {
        "imagePullPolicy": "Always",
        "image": "gcr.io/kubeflow-images-public/katib/v1alpha3/suggestion-goptuna"
      },
      "cmaes": {
        "imagePullPolicy": "Always",
        "image": "gcr.io/kubeflow-images-public/katib/v1alpha3/suggestion-goptuna"
      },
:
```

Reply:
Agree with @c-bata . User is not concerned about the internal implementation.

Reply:
Then we should update doc which implementation we currently support for which algorithm.
E.g random: hyperopt, chocolate, goptuna.